src/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -132,9 +132,7 @@ public ReadOnlySequence(ReadOnlyMemory<T> readOnlyMemory)
             }
             else if (typeof(T) == typeof(char))
             {
-                // TODO: MemoryMarshal.TryGetString -- doesn't exist?
-                // https://github.com/dotnet/corefx/issues/27451
-                MemoryExtensions.TryGetString(((ReadOnlyMemory<char>)(object)readOnlyMemory), out string text, out int start, out length);
+                MemoryMarshal.TryGetString(((ReadOnlyMemory<char>)(object)readOnlyMemory), out string text, out int start, out length);
```

Comment:
Do you need to handle the case where TryGetString returns false?

Reply:
I think you have to. This return false if readOnlyMemory was constructed from a char array instead of a string.

cc @pakrym, @davidfowl

Reply:
```C#
string str = "abc";
ReadOnlyMemory<char> memFromString = str.AsReadOnlyMemory();
if (memFromString.TryGetString(out string text, out _, out _)) // True
{
    Console.WriteLine(text);
}

char[] charArray = { '1', '2', '3' };
var memFromCharArray = new ReadOnlyMemory<char>(charArray);
if (memFromCharArray.TryGetString(out text, out _, out _)) // False
{
    Console.WriteLine(text);
}
```

Reply:
Its cause by the earlier `TryGetArray` switch?

Reply:
Maybe add an assert.